script.js

```diff
@@ -1,4 +1,435 @@
+  } else if (currentGameMode == "enter user name") {
+    myOutputValue = setUserName(input);
+  } else if (currentGameMode == "Choose your game") {
+    myOutputValue = startGame(input);
+  } else if (currentGameMode == "Normal") {
+    myOutputValue = normalGame(input);
+  } else if (currentGameMode == "Reversed") {
+    myOutputValue = reversedGame(input);
+  } else if (currentGameMode == "Korean") {
+    myOutputValue = koreanGame(input);
+  } else if (currentGameMode == "AI") {
+    myOutputValue = aiGame(input);
+  }
+  // console log to track which game mode we are current in
+  console.log("currentGameMode", currentGameMode);
+  return myOutputValue;
+};
+
+//----------- Set user name game mode --------------------
+
+var setUserName = function (playerName) {
+  userName = playerName;
+  currentGameMode = "Choose your game";
+  var myOutputValue = `Hello ${userName}! Please choose the game mode you'll like to play. <br><br>Type in "Normal", "Reversed", "Korean", "AI"`;
+  return myOutputValue;
+};
+
+// ----------- game mode to choose the type of game---------
+
+var startGame = function (input) {
+  var myOutputValue = `Please type "Normal", "Reversed", "Korean", "AI"`;
+  if (input == "Normal") {
+    currentGameMode = "Normal";
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br>This is the normal scissors, paper, stone game.<br><br>Start by typing in "scissors", "paper" or "stone"`;
+  }
+  if (input == "Reversed") {
+    currentGameMode = "Reversed";
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br>This is the REVERSED of scissors, paper, stone game.<br><br>Start by typing in "scissors", "paper" or "stone"`;
+  }
+  if (input == "Korean") {
+    currentGameMode = "Korean";
+    myOutputValue = `You have selected ${input} as your game of choice! <br><br>The Ultimate Winner is determined in a draw game.<br><br>The Ultimate Winner is the previous winner when the current game is a draw!<br><br>Start by typing in "scissors", "paper" or "stone"`;
+  }
+  if (input == "AI") {
+    currentGameMode = "AI";
+    myOutputValue = `You have selected Computer Vs AI as your game of choice! <br><br>An AI machine will pick for you!<br><br>Start by pressing Submit`;
+  }
+  return myOutputValue;
+};
+
+// ------------------ Normal game mode ----------------------
+
+var normalGame = function (input) {
+  // to generate computer's move. console log to see response and verify outcome.
+  var computerMove = randomItem();
+  console.log("computer", computerMove);
+
+  // to generate the icons for the moves
+  var iconComputer = iconLogo(computerMove);
+  console.log("iconComputer", iconComputer);
+  var iconPlayer = iconLogo(input);
+  console.log("iconPlayer", iconPlayer);
+
+  // to call the win, draw, invalid functions in main to help determine outcome.
+  var draw = drawGame(input, computerMove);
+  var win = playerWin(input, computerMove);
+  var tryAgain = invalid(input);
```

Comment:
Understand your thought process for decomposing here, it is good, however, I think it will be clearer to put these functions directly within the if statement (see below)